qiskit/circuit/library/pauli_evolution.py

```diff
@@ -122,7 +122,19 @@ class docstring for an example.
         if label is None:
             label = _get_default_label(operator)
 
-        num_qubits = operator[0].num_qubits if isinstance(operator, list) else operator.num_qubits
+        if isinstance(operator, list):
+            if len(operator) == 0:
+                raise ValueError("Operator cannot be represented using an empty list.")
```

Comment:
How about 
```suggestion
                raise ValueError("The argument 'operator' cannot be an empty list.")
```
I feel like "The operator cannot be represented" is somehow are more mathematical statement than we intend here 🙂